core/src/main/java/org/elasticsearch/search/aggregations/bucket/terms/TermsAggregationBuilder.java

```diff
@@ -196,7 +197,11 @@ public TermsAggregationBuilder order(Terms.Order order) {
         if (order == null) {
             throw new IllegalArgumentException("[order] must not be null: [" + name + "]");
         }
-        this.order = order;
+        if(order instanceof CompoundOrder || InternalOrder.isTermOrder(order)) {
+            this.order = order; // if order already contains a tie-breaker we are good to go
+        } else { // otherwise add a tie-breaker by using a compound order
+            this.order = Terms.Order.compound(order);
+        }
```

Comment:
No because that method calls this one again. Also this may result in a nested `CompoundOrder`, which doesn't make sense.